src/mango/test/26-test_query.py

```diff
@@ -0,0 +1,153 @@
+# Licensed under the Apache License, Version 2.0 (the "License"); you may not
+# use this file except in compliance with the License. You may obtain a copy of
+# the License at
+#
+#   http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+# WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+# License for the specific language governing permissions and limitations under
+# the License.
+
+import copy
+import mango
+import unittest
+import json
+from textwrap import dedent
+
+BOOLEN_ADB_DOC = {
+    "_id": "hoary_marmot",
+    "wiki_page": "http://en.wikipedia.org/wiki/Hoary_marmot",
+    "min_weight": 4,
+    "max_weight": 40,
+    "min_length": 0.2,
+    "max_length": 1,
+    "latin_name": "Marmota caligata",
+    "class": "mammal",
+    "diet": "herbivore",
+    "inherit_the_earth": True,
+    "likes": ["to take tea", "kittens", "trees"],
+    "habitat": [{"mountainous_environments": 8200, "states": ["Alaska", "Canada"]}],
+}
+
+BOOLEN_ADB_DOC2 = {
+    "_id": "hoary_marmots",
+    "wiki_page": "http://en.wikipedia.org/wiki/Hoary_marmot",
+    "min_weight": 3,
+    "max_weight": 30,
+    "min_length": 0.2,
+    "max_length": 1,
+    "latin_name": "Marmota caligatas",
+    "class": "mammal",
+    "diet": "carnivore",
+    "inherit_the_earth": True,
+    "likes": ["to take tea", "kittens", "trees", "games"],
+    "habitat": [{"mountainous_environments": 8200, "states": ["Alaska", "Canada"]}],
+}
+
+KOOKABURRA = {
+    "_id": "kookaburra",
+    "min_length": 0.28,
+    "max_length": 0.42,
+    "wiki_page": "http://en.wikipedia.org/wiki/Kookaburra",
+    "class": "bird",
+    "diet": "carnivore",
+    "latin_name": "Dacelo novaeguineae",
+}
+
+DDOC = {
+    "_id": "_design/views101",
+    "indexes": {
+        "animals": {
+            "index": dedent(
+                """\
+                function(doc){
+                  index(\"default\", doc._id);
+                  if(doc.min_length){
+                    index(\"min_length\", doc.min_length, {\"store\": \"yes\"});
```

Comment:
`{"store":"yes"}` looks a bit odd, I am not familiar with clouseau but I remember seeing `"store": true|false`, maybe confusing with Nouveau though...

Reply:
fixed in https://github.com/apache/couchdb/pull/5339/commits/ce1a450b38aae5022da54a846a15c66eab0182a9

Reply:
Thanks! I see https://docs.couchdb.org/en/stable/ddocs/search.html uses `true` vs `false` indicating `false` is default. Would it matter if we pass in `"true"` as a string and it would be promoted to `true`? Can the test check for the effect of `store`? Maybe it if doesn't matter in the test we can go with the default (`false`).